### PR TITLE
Converted legacy helper actions

### PR DESCRIPTION
**Pull request by dannyroosevelt**

### CODE REVIEW

**Comment by dylburger** [comment]
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/helper_functions/actions/schedule-task-in-future/schedule-task-in-future.mjs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    };
+    if (this.secret) {
+      headers["x-pd-secret"] = this.secret;
+    }
+
+    return await axios($, {
+      url: `${this.taskSchedulerURL}/schedule`,
+      headers,
+      data: {
+        timestamp: this.ts,
```
Comment (dylburger) [inline components/helper_functions/actions/schedule-task-in-future/schedule-task-in-future.mjs:49]: Does this work for you? i.e. can you actually reference `this.ts` after exporting it? That didn't work for me in a component code step.

Comment (dannyroosevelt) [reply components/helper_functions/actions/schedule-task-in-future/schedule-task-in-future.mjs:49]: Doesn't work for me either 
image

**Comment by dannyroosevelt** [comment]
It looks like when you declare `app` props directly in the action file, we use the `key` to name the action in the builder UI (ie, `helper_functions_end_workflow`), but if you import `helper_functions` from the app file, we'll use the `name` in the UI, which is much more user friendly. 

That's news to me -- @js07 can you plz adjust all the actions accordingly? I pushed a few slight tweaks to the PR FYI, so make sure to pull down the latest.

File: components/helper_functions/actions/schedule-task-in-future/schedule-task-in-future.mjs (hunk 1 above)
Comment (js07) [reply components/helper_functions/actions/schedule-task-in-future/schedule-task-in-future.mjs]: Fixed in https://github.com/PipedreamHQ/pipedream/pull/2297/commits/abdd137cd35aeac770fd3f9a532d0e0f483d997a

**Comment by js07** [comment]
I removed the `app` prop declaration and added a `helper_functions` prop imported from the app file, to the action files.

File: components/helper_functions/actions/end-workflow/end-workflow.mjs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      default: true,
+    },
+  },
+  async run({ $ }) {
+    if (!this.condition) {
+      $.export("$summary", "Workflow continued. Condition evaluated to false.");
+      return;
+    }
+    $.flow.exit(this.reason);
+    $.export("$summary", `Workflow ended. Reason: \`${this.reason}\``);
```
[the remark this answers is not in the document]
Comment (dannyroosevelt) [reply components/helper_functions/actions/end-workflow/end-workflow.mjs:31]: Oh, yea great catch and good idea

**Pull request merged**